apps/experiments/views/experiment.py

```diff
@@ -329,6 +336,8 @@ def form_valid(self, form, file_formset):
         if file_formset:
             files = file_formset.save(self.request)
             self.object.files.set(files)
+
+        self.object.set_tools(tool_names=self.request.POST.getlist("tools"))
```

Comment:
you could just make the form do this for you: https://github.com/dimagi/open-chat-studio/blob/6d68e88421dd60235e88b46b9c0d87b48129712c/apps/assistants/forms.py#L9

Even if you stick with the select box you could still have a field in the form and it would persist it to the model (I think)